### ISSUE TITLE
CookieManager return empty array on iOS

### ISSUE THREAD

**Issue by nickfaces** [KevinnZou/compose-webview-multiplatform#42]
I need to get auth cookie when page is loaded. 
I try it like this `cookies = state.cookieManager.getCookies("https://myurl/")`

It works fine on Android, but on iOS I get just `[]`

**Comment by KevinnZou** [comment]
Thank you for your feedback. Could you please provide more code that shows where you are attempting to call the `getCookies` function?

**Comment by nickfaces** [comment]
Yes. 

````
Scaffold {
            var state = rememberWebViewState(url )
            state.webSettings.apply {
                isJavaScriptEnabled = true
                customUserAgentString =
                    "Mozilla/5.0 (Linux; Android 8.0; Pixel 2 Build/OPD3.170816.012) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4577.82 Mobile Safari/537.36"
                androidWebSettings.apply {
                    isAlgorithmicDarkeningAllowed = true
                    safeBrowsingEnabled = true
                }

                LaunchedEffect(state.loadingState) {
                    if (state.loadingState is LoadingState.Finished) {
                        try {
                            val cookies  =  state.cookieManager.getCookies(url)
                        } catch (e: Exception) {
                           ///
                        }

                    }
                }

                WebView(
                    state = state,
                    modifier =
                    Modifier
                        .fillMaxSize(),
                    navigator = rememberWebViewNavigator()
                )
            }


```

**Comment by nickfaces** [comment]
I think I had something wrong with iOS version.
I am trying to add "Cookie" to additional headers of state. On Android I get an authorized site in WebView, but on iOS - no authorized.

**Comment by KevinnZou** [comment]
Thank you for your reply! I apologize for the inconvenience, but currently, additional headers are not functional on iOS. This issue was discovered recently and I will fix it by the end of the week.

**Comment by nickfaces** [comment]
Thx!  Issue  only applies to additional headers or Cookie too?

**Comment by KevinnZou** [comment]
Only additional headers will be fixed. Regarding the issue with the Cookie, it is not behaving as expected. It worked correctly during testing. Have you tested it with other URLs as well?

**Comment by musooff** [comment]
Same here ios cookies return empty.

**Comment by KevinnZou** [comment]
> Same here ios cookies return empty.

@musooff Thank you for your feedback! Could you please provide the code where you are trying to call the `getCookies` function and the URL? Have you tested it with other URLs?

**Comment by musooff** [comment]
it is not working with [naver.com](https://www.naver.com)
login in (or sign up) would be helpful in testing as it should return additional cookies. 
nid.naver.com <- sign in page for quick testing

**Comment by KevinnZou** [comment]
@musooff @nickfaces I believe I have identified the issue. On iOS, there is only a method available to retrieve all cookies. Therefore, I need to manually filter the cookies based on the URL. To do this, I use `url.contains(cookie.domain)` to determine if the cookie should be returned for the specific URL. 

However, I have noticed that most `cookie.domain` values begin with a single dot, causing `url.contains(cookie.domain)` to return false. As a result, the array returned on the iOS side is empty.

To resolve the issue, I will first remove the prefix dot from the domain and then use `url.contains()` to filter them. I will do this by using `url.contains(cookie.domain.removePrefix("."))`. After that change, the domain with a dot prefix will also be included in the return list.

I will fix the problem and merge it as soon as possible. I apologize for any inconvenience caused by this issue. Thank you again for your feedback.
